From 251a4eeba4fe0d96fb27f448202e66ba05310371 Mon Sep 17 00:00:00 2001
From: Fei Wang <fei.w.wang@intel.com>
Date: Mon, 31 Aug 2020 14:35:13 -0400
Subject: [PATCH 1/4] Mark's change width/height sbs.

---
 libavcodec/cbs_av1_syntax_template.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libavcodec/cbs_av1_syntax_template.c b/libavcodec/cbs_av1_syntax_template.c
index 19b82bc3f8..cf9c04f7d5 100644
--- a/libavcodec/cbs_av1_syntax_template.c
+++ b/libavcodec/cbs_av1_syntax_template.c
@@ -636,6 +636,14 @@ static int FUNC(tile_info)(CodedBitstreamContext *ctx, RWContext *rw,
             current->tile_rows_log2;
         current->tile_rows = (sb_rows + tile_height_sb - 1) / tile_height_sb;
 
+	for (i = 0; i < current->tile_cols - 1; i++)
+            infer(width_in_sbs_minus_1[i], tile_width_sb - 1);
+        infer(width_in_sbs_minus_1[i],
+              sb_cols - (current->tile_cols - 1) * tile_width_sb - 1);
+        for (i = 0; i < current->tile_rows - 1; i++)
+            infer(height_in_sbs_minus_1[i], tile_height_sb - 1);
+        infer(height_in_sbs_minus_1[i],
+              sb_rows - (current->tile_rows - 1) * tile_height_sb - 1);
     } else {
         int widest_tile_sb, start_sb, size_sb, max_width, max_height;
 
-- 
2.17.1

