From 3ae245b1ca3a6eef29c9fa01afcc7c291aadb850 Mon Sep 17 00:00:00 2001
From: Fei Wang <fei.w.wang@intel.com>
Date: Mon, 31 Aug 2020 13:38:14 -0400
Subject: [PATCH 2/4] avcodec/cbs_av1: add support for frame restoration type

Signed-off-by: Fei Wang <fei.w.wang@intel.com>
---
 libavcodec/av1.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libavcodec/av1.h b/libavcodec/av1.h
index 6c0e32485b..0f99ae4829 100644
--- a/libavcodec/av1.h
+++ b/libavcodec/av1.h
@@ -160,4 +160,12 @@ enum {
     AV1_SCALABILITY_L4T7_KEY_SHIFT = 28,
 };
 
+// Frame Restoration types (section 6.10.15)
+enum {
+    AV1_RESTORE_NONE       = 0,
+    AV1_RESTORE_WIENER     = 1,
+    AV1_RESTORE_SGRPROJ    = 2,
+    AV1_RESTORE_SWITCHABLE = 3,
+};
+
 #endif /* AVCODEC_AV1_H */
-- 
2.17.1

